config/src/vespa/config/frt/frtconnection.cpp

```diff
@@ -14,19 +14,21 @@ namespace config {
 
 FRTConnection::FRTConnection(const vespalib::string& address, FRT_Supervisor& supervisor, const TimingValues & timingValues)
     : _address(address),
+      _transientDelay(timingValues.transientDelay),
+      _fatalDelay(timingValues.fatalDelay),
       _supervisor(supervisor),
+      _lock(),
       _target(0),
       _suspendedUntil(),
       _suspendWarned(),
       _transientFailures(0),
-      _fatalFailures(0),
-      _transientDelay(timingValues.transientDelay),
-      _fatalDelay(timingValues.fatalDelay)
+      _fatalFailures(0)
 {
 }
 
 FRTConnection::~FRTConnection()
 {
+    std::lock_guard guard(_lock);
```

Comment:
this does not seem like it helps...
(any conflicting thread that gets the lock after us will try to lock the mutex after it has been destructed)

Reply:
That was only intended to ensure thread visibility.

Reply:
If you need to ensure thread visibility here, it kind of implies that there is no synchronization between using the object and deleting it, which could lead to locking the deleted mutex. In other words; if the locking was added to make TSAN happy, this will probably not fix the race, just hide it.